bin/bench-transaction/src/time_counting_benchmarks/prove.rs

```diff
@@ -120,8 +120,9 @@ fn core_benchmarks(c: &mut Criterion) {
 
 fn prove_transaction(executed_transaction: ExecutedTransaction) -> Result<()> {
     let executed_transaction_id = executed_transaction.id();
+    let (_, _, tx_witness, _, input_notes) = executed_transaction.into_parts();
     let proven_transaction: ProvenTransaction =
-        LocalTransactionProver::default().prove(executed_transaction.into())?;
+        LocalTransactionProver::default().prove(tx_witness, input_notes)?;
```

Comment:
@bobbinth @PhilippGackstatter The current diffs show what it would look like if we simply took input notes out of `TransactionInputs` struct.

Noting 
- All fns that use `TransactionInputs` also now require `InputNotes<InputNote>` to be provided separately
- The note checker no longer needs to mutate `TransactionInputs` to execute transactions with different combinations of notes
- As @PhilippGackstatter mentioned offline, an alternate approach could be to keep input notes in `TransactionInputs` and instead create another type to be returned by `fn get_transaction_inputs()` which currently returns a tuple that is basically `TransactionInputs` minus the input notes.